### PR TITLE
P-2564 Modal component

### PR DESCRIPTION
**Pull request by gunjankothari**

### CODE REVIEW

File: packages/core/src/mappings/Modal.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  name: "Modal",
+  tag: "Modal",
+  description: "A basic Modal Component",
+  category: COMPONENT_TYPES.container,
+  props: {
+    animationType: {
+      group: GROUPS.advanced,
+      name: "animationType",
+      label: "animationType",
+      description: "Animatio Type",
```
Comment (dhruska) [inline packages/core/src/mappings/Modal.js:19]:
```suggestion
      description: "Animation Type",
```

File: packages/core/src/mappings/Modal.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    transparent: createBoolProp({
+      group: GROUPS.basic,
+      label: "Transparent",
+      description: "Should be Transparent",
+    }),
+    visible: createBoolProp({
+      group: GROUPS.data,
+      label: "Visible",
+      description: "Should be Visible",
```
Comment (dhruska) [inline packages/core/src/mappings/Modal.js:35]:
```suggestion
      description: "Determines whether the modal is visible",
```

File: packages/core/src/mappings/Modal.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      editable: false,
+      required: false,
+      formType: FORM_TYPES.flatArray,
+      propType: PROP_TYPES.STRING,
+      defaultValue: "none",
+    },
+    transparent: createBoolProp({
+      group: GROUPS.basic,
+      label: "Transparent",
+      description: "Should be Transparent",
```
Comment (dhruska) [inline packages/core/src/mappings/Modal.js:30]: (copied from docs)
```suggestion
      description: "Determines whether the modal will fill the entire view. Setting this to true will render the modal over a transparent background",
```

File: packages/core/src/mappings/Modal.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+} from "@draftbit/types";
+
+export const SEED_DATA = {
+  name: "Modal",
+  tag: "Modal",
+  description: "A basic Modal Component",
+  category: COMPONENT_TYPES.container,
+  props: {
+    animationType: {
+      group: GROUPS.advanced,
```
Comment (dhruska) [inline packages/core/src/mappings/Modal.js:16]: I already merged this, but I'm not actually seeing this prop anywhere. `Advanced` may be hidden, currently, so I think this needs to be `basic`.

Comment (gunjankothari) [reply packages/core/src/mappings/Modal.js:16]: OK. I will create another PR to make them basic.

**Pull request merged**